org.opentosca.container.engine.plan.plugin.bpel/src/org/opentosca/container/engine/plan/plugin/bpel/BpelPlanEnginePlugin.java

```diff
@@ -140,6 +141,155 @@ public String getLanguageUsed() {
         return capabilities;
     }
 
+    
+    public boolean deployPlanFile(final Path filePath, final CSARID csarId, Map<String,String> endpointMetadata) {
+        List<File> planContents;
+        File tempDir;
+        File tempPlan;
+        QName portType = null;
+
+        if (this.fileAccessService != null) {
+            // creating temporary dir for update
+            tempDir = this.fileAccessService.getTemp();
+            tempPlan = new File(tempDir, filePath.getFileName().toString());
+            BpelPlanEnginePlugin.LOG.debug("Unzipping Plan '{}' to '{}'.", filePath.getFileName().toString(),
+                                           tempDir.getAbsolutePath());
+            planContents = this.fileAccessService.unzip(filePath.toFile(), tempDir);
+        } else {
+            BpelPlanEnginePlugin.LOG.error("FileAccessService is not available, can't create needed temporary space on disk");
+            return false;
+        }
+        
+     // changing endpoints in WSDLs
+        ODEEndpointUpdater odeUpdater;
+        try {
+            odeUpdater = new ODEEndpointUpdater(SERVICESURL, ENGINE);
+            portType = odeUpdater.getPortType(planContents);
+            if (!odeUpdater.changeEndpoints(planContents, csarId)) {
+                BpelPlanEnginePlugin.LOG.error("Not all endpoints used by the plan {} have been changed",
+                                               filePath);
+            }
+        }
+        catch (final WSDLException e) {
+            BpelPlanEnginePlugin.LOG.error("Couldn't load ODEEndpointUpdater", e);
+        }
+
+        // update the bpel and bpel4restlight elements (ex.: GET, PUT,..)
+        BPELRESTLightUpdater bpelRestUpdater;
+        try {
+            bpelRestUpdater = new BPELRESTLightUpdater();
+            if (!bpelRestUpdater.changeEndpoints(planContents, csarId)) {
+                // we don't abort deployment here
+                BpelPlanEnginePlugin.LOG.warn("Could'nt change all endpoints inside BPEL4RESTLight Elements in the given process {}", filePath);
+            }
+        }
+        catch (final TransformerConfigurationException e) {
+            BpelPlanEnginePlugin.LOG.error("Couldn't load BPELRESTLightUpdater", e);
+        }
+        catch (final ParserConfigurationException e) {
```

Comment:
could be merged into preceeding catch clause

Reply:
Changed log messages to make it more distinguishable